lib/Conversion/TorchOnnxToTorch/DefaultDomainGtoP.cpp

```diff
@@ -591,6 +591,355 @@ void mlir::torch::onnx_c::populateDefaultDomainGtoP(
+        Type inpIntDType = inputIntType.getDtype();
+        Type si32Ty = rewriter.getIntegerType(32, true);
+        Type f32Ty = rewriter.getF32Type();
+        Type i1Ty = rewriter.getI1Type();
+
+        // Value constants
+        Value noneConst = b.create<Torch::ConstantNoneOp>();
+        Value negTwoConst =
+            b.create<Torch::ConstantIntOp>(rewriter.getI64IntegerAttr(-2));
+        Value negOneConst =
+            b.create<Torch::ConstantIntOp>(rewriter.getI64IntegerAttr(-1));
+        Value zeroConst =
+            b.create<Torch::ConstantIntOp>(rewriter.getI64IntegerAttr(0));
+        Value oneConst =
+            b.create<Torch::ConstantIntOp>(rewriter.getI64IntegerAttr(1));
+        Value twoConst =
+            b.create<Torch::ConstantIntOp>(rewriter.getI64IntegerAttr(2));
+        Value sixConst = b.create<Torch::ConstantIntOp>(
+            rewriter.getI64IntegerAttr(6)); // dtype float32
+
+        VTType dftLenType = VTType::get(ctx, unranked, inpIntDType);
+        Type freqBinsIntType = VTType::get(ctx, shapeNMBp2, si32Ty);
+        Type freqBinsFltType = VTType::get(ctx, shapeNMBp2, f32Ty);
+
+        Value dftLengthDivTwoFlt =
+            b.create<Torch::AtenDivIntOp>(dftLengthItem, twoConst);
+        Value dftLengthDivTwo =
+            b.create<Torch::AtenIntFloatOp>(dftLengthDivTwoFlt);
+        Value numSpectrogramBins =
+            b.create<Torch::AtenAddIntOp>(dftLengthDivTwo, oneConst);
+        Value numSpectrogramBinsItem = numSpectrogramBins;
+        Value freqBinsInit = b.create<Torch::AtenArangeOp>(
+            freqBinsIntType, numMelBinsItem, /*dtype=*/sixConst,
+            /*layout=*/noneConst, /*device=*/noneConst,
+            /*pin_memory=*/noneConst);
+
+        // convert input Freq Hz to Mel
+        Value twoFiveNineFiveConst =
+            b.create<Torch::ConstantFloatOp>(rewriter.getF64FloatAttr(2595));
+        Value sevenHConst =
+            b.create<Torch::ConstantFloatOp>(rewriter.getF64FloatAttr(700));
+        Value tenConst =
+            b.create<Torch::ConstantFloatOp>(rewriter.getF64FloatAttr(10));
+
+        Value lfDiv7Hfloat = b.create<Torch::AtenDivFloatOp>(
+            lowerEdgeHzItem,
+            sevenHConst); // Since all others need tensor to operate
+        Type freqType = VTType::get(ctx, unranked, inpFpDType);
+        Value lfDiv7H =
+            b.create<Torch::PrimNumToTensorScalarOp>(freqType, lfDiv7Hfloat);
+        Value lfDiv7HAdd1 = b.create<Torch::AtenAddScalarOp>(
+            freqType, lfDiv7H, oneConst, /*alpha =*/oneConst);
+        Value lfDiv7HAdd1Log10 =
+            b.create<Torch::AtenLog10Op>(freqType, lfDiv7HAdd1);
+        Value lfMel = b.create<Torch::AtenMulScalarOp>(
+            freqType, lfDiv7HAdd1Log10, twoFiveNineFiveConst);
+
+        Value hfDiv7Hfloat = b.create<Torch::AtenDivFloatOp>(
+            upperEdgeHzItem,
+            sevenHConst); // Since all others need tensor to operate
```

Comment:
Similarly here

Reply:
This comment can actually be removed. It was something that was useful to track which operations were tensor based and which ones worked on floats/ints.